Xamarin.Forms.Core.UnitTests/TemplatedViewUnitTests.cs

```diff
@@ -27,5 +27,31 @@ public ExpectedView()
             {
             }
         }
+
+		public class MyTemplate : StackLayout
+		{
+			public MyTemplate()
+			{
+				Children.Add(new ContentPresenter());
+			}
+		}
+
+		[Test]
+		public void BindingsShouldBeAppliedOnTemplateChange()
+		{
+			var template0 = new ControlTemplate(typeof(MyTemplate));
+			var template1 = new ControlTemplate(typeof(MyTemplate));
+			var label = new Label();
+			label.SetBinding(Label.TextProperty, ".");
+			var cv = new ContentView {
+				ControlTemplate = template0,
+				Content = label
+			};
+			cv.BindingContext = "Foo";
+
+			Assume.That(label.Text, Is.EqualTo("Foo"));
```

Comment:
Cool. So you use an assume to indicate this is not the test but an invariant of the the test itself.